### DISCUSSION TITLE
PlanetScale connection without a `shadowDatabaseUrl` is not working

### DISCUSSION THREAD

**Discussion by mohsen1** [prisma/prisma#15087] [Q&A] [answered]
Getting the `prisma migrate could not create the shadow database` error. 

<details>
<summary><code>schema.prisma</code></summary>

```

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

model Post {
  id        Int      @default(autoincrement()) @id
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  title     String   @db.VarChar(255)
  text      String   @db.Text()
}


```

</details>

Since PlanetScale does not allow more than 2 branches on free plan, this is stopping people from trying things out with deploying a toy project and also having a dev environment

**Comment by janpio** [comment, marked as answer]
Yes, that is an unfortunate limitation in combination with the requirement to manually set a shadow database URL for `migrate dev`. But Planetscale has its own Migration workflow anyway, so we generally do not recomment using `migrate dev` there. You can read more about that here: https://www.prisma.io/docs/guides/database/using-prisma-with-planetscale#:~:text=Making%20schema%20changes,branch%20is%20merged.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
